articles/article-088-no-free-lunch.md

```diff
@@ -0,0 +1,25 @@
+<!-- BEGIN TITLE -->
+# There is no such thing as a free lunch
+<!-- END TITLE -->
+
+<!-- BEGIN BODY -->
+In Computer Science we have  a wide range of very different problems and accordingly many approaches to tackle them.
+We might for example try to find some 'optimal' solution for a search problem, but it is unclear which specific algorithm one should use.
+Hence, it is only natural to think about algorithms that could always provide us the best solution, regardless of the problems subtleties.
+
+However, even though a method might result in higher performance for one problem, this comes with the price of diminishing performance for other problems.
```

Comment:
Even though an algorithm performs better for one problem, it may have a worse performance with for other applicable problems. 
- I put applicable because no algorithm can be used for all problems

Reply:
That sentence would be wrong (the 'may' is on the wrong side), since it is guaranteed that it will have a worse performance for other problems if it works better for one.

Reply:
Also I feel as if the link would be missing without an however.

Reply:
The however and even though combination makes it look like you are saying: But, but my point is... you can drop the may and change the second part to be "it will perform worse with other problems".